### ISSUE TITLE
$post bug

### ISSUE THREAD

**Issue by ghost** [littlebizzy/hovercraft#2]
bug at:

https://github.com/littlebizzy/hovercraft/blob/4ffa85d1b8f6bef0b8583eb3cc4653fd0e6311f8/inc/css-generator.php#L7

error log:
![Screenshot_24](https://user-images.githubusercontent.com/108953087/210408870-78cb89f8-ad8f-4c03-922c-0c58923c0749.png)

**Comment by jessuppi** [comment]
Thanks for reporting @yamanmucahit 

I'm not sure but guessing this error is triggered when content is loaded that does not have a $post ID... maybe there needs to be an `if` statement or conditions added to this variable.

**Comment by ghost** [comment]
hey @jessuppi 
i did add an if statement (#4) and it was fixed but seems i have bigger problems as in content without id lol.
if (!commit(#4)) {
closeissue() && echo 'thanks'
}

**Comment by jessuppi** [comment]
Hmm, shouldn't it be the opposite, e.g. `if(isset...` without the exclamation point?

```
if(!isset($post->ID)) {
	$url_featured_image = wp_get_attachment_url( get_post_thumbnail_id($post->ID), 'large' );
}
```

**Comment by ghost** [comment]
yes, my bad. corrected in the #5.

**Comment by jessuppi** [comment]
I'm not sure if this resolves the issue entirely, because it doesn't take into consideration:

```
if (!empty( $url_featured_image )) { 
    $hero_image = $url_featured_image; 
} else { 
    $hero_image = $url_header_image; 
}
```

**Comment by ghost** [comment]
you're right. actually it did not solve it to begin with. turns out letting the parameter default is best here. since the default is 'global $post' it actually sets the variable. something like:
`$url_featured_image = wp_get_attachment_url( get_post_thumbnail_id(), 'large' );`
so using
`$url_featured_image = wp_get_attachment_url( get_post_thumbnail_id(), 'large' );
echo $url_featured_image;`
returns the url finally.

**Comment by ghost** [comment]
solved in #19. closing issue.
